lib/libmap.c

```diff
@@ -0,0 +1,67 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <time.h>
+#include "libmap.h"
+
+#define printColoredChar(color) (printf("\033[38;5;%dm██\033[0m", color))
+
+void printLine(color pixels[WIDTH], int lineNumber)
+{
+    printf("\033[1;38;5;8m %d", lineNumber);
+    if (lineNumber < 10)
+        printf(" ");
+    for (int i = 0; i < WIDTH; i++)
+        printColoredChar(pixels[i]);
+    printf("\033[1;38;5;8m %d", lineNumber);
+    printf("\n");
+}
+// get random color of the enum
+
+color getRandomColor()
+{
+    return (color)(rand() % NB_COLORS);
+}
+
+void printColumNumber()
+{
+    printf("\033[51;1;38;5;8m   ");
+    for (int i = 1; i <= WIDTH; i++)
+    {
+        if (i < 10)
+            printf(" ");
+        printf("%d", i);
+    }
+    printf("\n");
+}
+
+void printMap(color map[HEIGHT][WIDTH])
+{
+    printColumNumber();
+    for (int i = 0; i < HEIGHT; i++)
+        printLine(map[i], i + 1);
+    printColumNumber();
+}
+
+void exportMap(color map[HEIGHT][WIDTH], char *filename)
```

Comment:
doesn't need to provide the filename as a parameter because it should be a constant